src/test/java/com/jcabi/http/wire/LastModifiedCashingWireTest.java

```diff
@@ -0,0 +1,117 @@
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.http.wire;
+
+import com.jcabi.aspects.Tv;
+import com.jcabi.http.Request;
+import com.jcabi.http.mock.MkAnswer;
+import com.jcabi.http.mock.MkContainer;
+import com.jcabi.http.mock.MkGrizzlyContainer;
+import com.jcabi.http.mock.MkQuery;
+import com.jcabi.http.request.JdkRequest;
+import com.jcabi.http.response.RestResponse;
+import java.net.HttpURLConnection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.hamcrest.core.IsAnything;
+import org.junit.Test;
+
+/**
+ * Test case for {@link LastModifiedCashingWire}.
+ * @author Igor Piddubnyi (igor.piddubnyi@gmail.com)
+ * @version $Id$
+ * @since 1.0
+ */
+public class LastModifiedCashingWireTest {
+
+    /**
+     * Time long ago.
+     */
+    public static final String LONG_AGO = "Wed, 15 Nov 1995 04:58:08 GMT";
+
+    /**
+     * LastModifiedCashingWire can cache GET requests.
+     * @throws Exception If fails
+     */
+    @Test
+    public final void cachesGetRequest() throws Exception {
+        final Map<String, String> headersCache = new ConcurrentHashMap<>();
+        headersCache.put(LastModifiedCashingWire.LAST_MODIFIED, LONG_AGO);
+        final MkContainer container = new MkGrizzlyContainer()
+            .next(
+                new MkAnswer.Simple(
+                    HttpURLConnection.HTTP_OK,
+                    headersCache.entrySet(),
+                    new byte[0]
+                )
+            )
+            .next(
+                new MkAnswer.Simple(
+                    HttpURLConnection.HTTP_NOT_MODIFIED
+                ),
+                new IsAnything<MkQuery>(),
+                Tv.TEN
+            ).start();
+        final Request req = new JdkRequest(container.home())
+            .through(LastModifiedCashingWire.class);
+        req.fetch().as(RestResponse.class)
```

Comment:
@piddubnyi this does the same thing that is done in the for loop below. seems unnecessary to me.